File: test-suite/python/security/cookie_security_clean.py
class Response:
    def set_cookie(self, name, value, **kwargs):
        self.cookie = (name, value, kwargs)

    def set_signed_cookie(self, name, value, **kwargs):
        self.signed_cookie = (name, value, kwargs)


class App:
    def __init__(self):
        self.config = {}


app = App()
response = Response()


SESSION_COOKIE_SECURE = True
CSRF_COOKIE_SECURE = True
SESSION_COOKIE_HTTPONLY = True
CSRF_COOKIE_HTTPONLY = True
SESSION_COOKIE_SAMESITE = "None"
CSRF_COOKIE_SAMESITE = "None"

app.config["SESSION_COOKIE_SECURE"] = True
app.config["SESSION_COOKIE_HTTPONLY"] = True
app.config["SESSION_COOKIE_SAMESITE"] = "Strict"

app.config.update(
    SESSION_COOKIE_SECURE=True,
    CSRF_COOKIE_SECURE=True,
    SESSION_COOKIE_SAMESITE="None",
)

app.config.from_mapping(
    SESSION_COOKIE_SECURE=True,
    SESSION_COOKIE_HTTPONLY=True,
    SESSION_COOKIE_SAMESITE="None",
)

response.set_cookie("sid", "value", secure=True, httponly=True, samesite="Lax")
response.set_cookie("oauth_state", "value", secure=True, httponly=True, samesite="None")
response.set_signed_cookie("signed", "value", secure=True, httponly=True, samesite="Strict")

File: test-suite/js/security/cookie-security-clean.ts
import { NextResponse } from "next/server";
import cookie from "cookie";

type ResponseLike = {
  cookie(name: string, value: string, options?: Record<string, unknown>): void;
  setHeader(name: string, value: string): void;
};

export function expressSafeSession(res: ResponseLike, sessionId: string): void {
  res.cookie("session_id", sessionId, {
    httpOnly: true,
    secure: true,
    sameSite: "lax",
  });
}

export function expressSafeCrossSite(res: ResponseLike, token: string): void {
  res.cookie("auth_token", token, {
    httpOnly: true,
    secure: true,
    sameSite: "none",
  });
}

export function nextResponseSafe(token: string): NextResponse {
  const response = NextResponse.json({ ok: true });
  response.cookies.set("refresh_token", token, {
    httpOnly: true,
    secure: true,
    sameSite: "strict",
  });
  return response;
}

export function nextResponseObjectFormSafe(token: string): NextResponse {
  const response = NextResponse.json({ ok: true });
  response.cookies.set({
    name: "login_token",
    value: token,
    httpOnly: true,
    secure: true,
    sameSite: "none",
  });
  return response;
}

export function rawSetCookieHeader(res: ResponseLike, jwt: string): void {
  res.setHeader("Set-Cookie", `jwt=${jwt}; Path=/; HttpOnly; Secure; SameSite=Lax`);
}

export function serializedCookie(token: string): string {
  return cookie.serialize("access_token", token, {
    httpOnly: true,
    secure: true,
    sameSite: "lax",
  });
}
